src/Microsoft.Extensions.ML/ModelLoaders/FileModelLoader.cs

```diff
@@ -105,13 +105,37 @@ public override ITransformer GetModel()
             return _model;
         }
 
+        public bool IsFileLocked(string filePath)
+        {
```

Comment:
This seems okay, but it is better to return the file handle directly instead of returning bool and trying to reopen the file. 
See the second answer here for a better solution: [https://stackoverflow.com/questions/50744/wait-until-file-is-unlocked-in-net](https://stackoverflow.com/questions/50744/wait-until-file-is-unlocked-in-net)